input/input.c

```diff
@@ -872,18 +875,70 @@ void mp_input_set_mouse_pos_artificial(struct input_ctx *ictx, int x, int y)
                 queue_remove(&ictx->cmd_queue, tail);
                 talloc_free(tail);
             }
-            mp_input_queue_cmd(ictx, cmd);
+            queue_cmd(ictx, cmd);
         }
     }
+}
+
+void mp_input_set_mouse_pos_artificial(struct input_ctx *ictx, int x, int y)
+{
+    input_lock(ictx);
+    set_mouse_pos(ictx, x, y);
+    input_unlock(ictx);
+}
+
+void mp_input_set_mouse_pos(struct input_ctx *ictx, int x, int y)
+{
+    input_lock(ictx);
+    if (ictx->opts->enable_mouse_movements)
+        set_mouse_pos(ictx, x, y);
     input_unlock(ictx);
 }
 
+static bool test_mouse(struct input_ctx *ictx, int x, int y, int rej_flags)
+{
+    bool res = false;
+    for (int i = 0; i < ictx->num_active_sections; i++) {
+        struct active_section *as = &ictx->active_sections[i];
+        if (as->flags & rej_flags)
+            continue;
+        struct cmd_bind_section *s = get_bind_section(ictx, bstr0(as->name));
+        if (s->mouse_area_set && test_rect(&s->mouse_area, x, y)) {
+            res = true;
+            break;
+        }
+    }
+    return res;
+}
+
+static bool test_mouse_active(struct input_ctx *ictx, int x, int y)
+{
+    return test_mouse(ictx, x, y, MP_INPUT_ALLOW_HIDE_CURSOR);
+}
+
+bool mp_input_test_mouse_active(struct input_ctx *ictx, int x, int y)
+{
+    input_lock(ictx);
+    bool res = test_mouse_active(ictx, x, y);
+    input_unlock(ictx);
+    return res;
+}
+
+bool mp_input_test_dragging(struct input_ctx *ictx, int x, int y)
+{
+    input_lock(ictx);
+    bool r = !ictx->opts->allow_win_drag ||
```

Comment:
nit, proably `bool res` for consistency with the above functions would be better.

EDIT: I see it is copied form previous code, so can be left as is if you prefer.